### PR TITLE
feat(components/layout): tokenize box styling

### PR DESCRIPTION
**Pull request by Blackbaud-ErikaMcVey**

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2774/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/forms-storybook/)
- [help-inline-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/help-inline-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/2774/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2774/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2774/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2774/playground/)

File: libs/components/layout/src/lib/modules/box/box-header.component.scss [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+  --sky-override-box-header-padding-right: var(--sky-margin-inline-xl);
+}
+
+@include compatMixins.sky-modern-overrides('sky-box') {
+  --sky-override-box-header-padding-right: var(--sky-margin-inline-xl);
+}
+
 sky-box-header {
   flex: 1 0;
-  order: 0;
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/layout/src/lib/modules/box/box-header.component.scss:3]: order wasn't doing anything here--they're already (and will always be) in that order

File: libs/components/layout/src/lib/modules/box/box.component.html [hunk 2]
```diff
@@ -10,52 +10,56 @@
...
-      @switch (headingLevel) {
-        @case (3) {
-          <h3
-            [id]="headerId"
-            [class]="headingClass + ' sky-margin-inline-xs'"
-            [class.sky-screen-reader-only]="headingHidden"
-          >
-            {{ headingText }}
-          </h3>
+      <div class="sky-box-header-right">
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/layout/src/lib/modules/box/box.component.html:13]: these changes fix how headers wrap in easy mode

File: libs/components/layout/src/lib/modules/box/box.component.scss [hunk 3]
```diff
@@ -43,20 +47,29 @@ sky-box {
...
     h3,
     h4,
     h5 {
+      display: inline;
       margin-block: 0;
     }
   }
 
   sky-box-content {
-    flex: 0 1 100%;
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/layout/src/lib/modules/box/box.component.scss:55]: this element is `display: blox` so setting `flex` on it has no effect

File: libs/components/layout/src/lib/modules/box/box.component.html [hunk 4]
```diff
@@ -10,52 +10,56 @@
...
+        @switch (headingLevel) {
+          @case (3) {
+            <!-- display: inline -->
```
Comment (Blackbaud-TrevorBurch) [inline libs/components/layout/src/lib/modules/box/box.component.html:16]: Are these comments still needed?

Comment (Blackbaud-ErikaMcVey) [reply libs/components/layout/src/lib/modules/box/box.component.html:16]: yes--they prevent prettier from formatting the heading elements in the way that adds extra space before and after. It is an alternative to using `skyTrim`

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Pull request merged**